include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+    static constexpr const char *log_level_debug2 = "DEBUG2";
+
+    /// The "annotations" setting.
+    std::map<std::string, std::string> annotations = {};
+
+    /// The "disabled_events" setting.
+    std::vector<std::string> disabled_events = {};
+
+    /// The "inputs" setting.
+    std::vector<std::string> inputs = {};
+
+    /// The "input_filepaths" setting.
+    std::vector<std::string> input_filepaths = {};
+
+    /// The "log_filepath" setting.
+    std::string log_filepath = {};
+
+    /// The "log_level" setting.
+    std::string log_level = log_level_err;
+
+    /// The "output_filepath" setting.
+    std::string output_filepath = {};
+
+    /// The "bouncer_base_url" setting.
+    std::string bouncer_base_url = "https://bouncer.ooni.io";
+
+    /// The "collector_base_url" setting.
+    std::string collector_base_url = {};
+
+    /// The "dns/nameserver" setting.
+    std::string dns_nameserver = {};
+
+    /// The "dns/engine" setting.
+    std::string dns_engine = "system";
+
+    /// The "geoip_asn_path" setting.
+    std::string geoip_asn_path = {};
+
+    /// The "geoip_country_path" setting.
+    std::string geoip_country_path = {};
+
+    /// The "ignore_bouncer_error" setting.
+    bool ignore_bouncer_error = true;
+
+    /// The "ignore_open_report_error" setting.
+    bool ignore_open_report_error = true;
+
+    /// The "max_runtime" setting.
+    double max_runtime = -1.0;
+
+    /// The "net/ca_bundle_path" setting.
+    std::string net_ca_bundle_path = {};
+
+    /// The "net/timeout" setting.
+    double net_timeout = 10.0;
+
+    /// The "no_bouncer" setting.
+    bool no_bouncer = false;
+
+    /// The "no_collector" setting.
+    bool no_collector = false;
+
+    /// The "no_asn_lookup" setting.
+    bool no_asn_lookup = false;
+
+    /// The "no_cc_lookup" setting.
+    bool no_cc_lookup = false;
+
+    /// The "no_ip_lookup" setting.
+    bool no_ip_lookup = false;
+
+    /// The "no_file_report" setting.
+    bool no_file_report = false;
+
+    /// The "no_resolver_lookup" setting.
+    bool no_resolver_lookup = false;
+
+    /// The "probe_asn" setting.
+    std::string probe_asn = {};
+
+    /// The "probe_cc" setting.
+    std::string probe_cc = {};
+
+    /// The "probe_ip" setting.
+    std::string probe_ip = {};
+
+    /// The "randomize_input" setting.
+    bool randomize_input = true;
+
+    /// The "save_real_probe_asn" setting.
+    bool save_real_probe_asn = true;
+
+    /// The "save_real_probe_cc" setting.
+    bool save_real_probe_cc = true;
+
+    /// The "save_real_probe_ip" setting.
+    bool save_real_probe_ip = false;
+
+    /// The "save_real_resolver_ip" setting.
+    bool save_real_resolver_ip = true;
+
+    /// The "software_name" setting.
+    std::string software_name = {};
+
+    /// The "software_version" setting.
+    std::string software_version = {};
+};
+
+/// Settings of the "CaptivePortal" nettest.
+class CaptivePortalSettings : public CommonSettings {
```

Comment:
In the FFI API, one needs to "know" what are nettest specific settings. Here there is a class for each nettest that contains nettest specific settings where available. This should help people understand what are the available settings for each given test. Inheritance is used such that one can pass a nettest-specific settings class to some common code that initialises all the common settings.